extensions/kubectl-cli/src/extension.ts

```diff
@@ -337,7 +337,44 @@ async function postActivate(
   let releaseToInstall: KubectlGithubReleaseArtifactMetadata | undefined;
   let releaseVersionToInstall: string | undefined;
   let currentVersion = kubectl.version;
-
+  // check if there is a new version to be installed and register the updater
+  let releaseToUpdateTo: KubectlGithubReleaseArtifactMetadata | undefined;
+  let releaseVersionToUpdateTo: string | undefined;
+  const latestAsset = await kubectlDownload.getLatestVersionAsset();
```

Comment:
I think there is a problem here. getLatestVersioAsset call can throw exception for various reasons:
1. github API is down
2. network is down
3. github API rate limit exceeded

In this implementation that would lead to cli tool update is not registered at all. To get out of it restart would be required.

Reply:
I've added try catch and checks for not initializes `latestAsset`. It prevents from failure to set updater and installer in case of  problems listed above and let at least use `Install`, `Upgrade/Downgrade` without need to restart podman desktop.